### PR TITLE
WIP: Add gif command

### PR DESCRIPTION
**Pull request by nehayward**

### CODE REVIEW

File: commands/gif_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+func TestGif(t *testing.T) {
+	w := bytes.NewBuffer(nil)
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Method, "GET")
+		assert.Equal(t, r.URL.Path, "/search")
+
+		w.Write([]byte(googleGifHTML))
```
Comment (zpatrick) [inline commands/gif_test.go]: we should be using an API to get these results, not endpoints that yield html.

File: commands/gif.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+func newGoogleRequest(url string, searchTerm string) (*http.Request, error) {
+	url = strings.TrimSuffix(url, "/")
+	url += "/search?tbm=isch&tbs=itp:animated&q=" + searchTerm
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", "Mozilla/5.0 (iPhone; U; CPU iPhone OS 4_0 like Mac OS X; en-us) AppleWebKit/532.9 (KHTML, like Gecko) Versio  n/4.0.5 Mobile/8A293 Safari/6531.22.7")
```
[the remark this answers is not in the document]
Comment (nehayward) [reply commands/gif.go]: It makes it so we can actually grab the urls.

File: commands/gif.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Status: %s, Code: %d", resp.Status, resp.StatusCode)
+	}
+
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
```
[the remark this answers is not in the document]
Comment (nehayward) [reply commands/gif.go]: My bad.

File: commands/gif.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+			}
+
+			// TODO: Return help gif or funny one
+			return fmt.Errorf("Searchquery is required")
+		},
+	}
+}
+
+func lookupGif(w io.Writer, client *GoogleClient, req *http.Request) error {
+	resp, err := client.Client.Do(req)
```
[the remark this answers is not in the document]
Comment (nehayward) [reply commands/gif.go]: Yeah I just had some confusion on being able to just deal with the body.

File: commands/gif.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Status: %s, Code: %d", resp.Status, resp.StatusCode)
+	}
+
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	bodyString := string(bodyBytes)
+
+	var regexFindContent = regexp.MustCompile(`var u=\'.*?(http.*?)\'`)
+	var regexFindURL = regexp.MustCompile(`http.*\w`)
```
[the remark this answers is not in the document]
Comment (nehayward) [reply commands/gif.go]: This allows us to not have to use a token, like we discussed. This my warrant a longer conversation.

File: models/gif.go [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package models
+
+type Gif struct {
```
[the remark this answers is not in the document]
Comment (nehayward) [reply models/gif.go]: I'll move it.

**Comment by zpatrick** [comment]
see my comments on: https://github.com/quintilesims/slackbot/issues/10
we can re-open when it is a good time to switch over to using the google api

**Pull request closed without merging**